presto-native-execution/presto_cpp/main/TaskManager.cpp

```diff
@@ -894,6 +894,9 @@ folly::Future<std::unique_ptr<Result>> TaskManager::getResults(
               maxSize,
               *bufferManager_);
         }
+        // Keep the promiseHolder alive to make sure that maxWaitMicros is
+        // effective.
+        keepPromiseAlive(promiseHolder, state);
```

Comment:
@icejoywoo The change looks reasonable to me. Though I would like to better understand the condition under which the promise is broken and response is returned immediately.

From what I understand If the task is running the promise should remain alive through the `getData` callback. 

If the task is started, but is not running or finished it should be in either one for these states: `kCanceled, kAborted, kFailed `

The `kAborted` and `kFailed` states should in theory be handled above (* there's a small chance that it may be caught during transition here, but the next request should return with a correct delay).

Do you see an immediate responses only when a task is "cancelled"? Or do you see this behaviour in general (for other task states?)